src/common/hashtable/src/keys_ref.rs

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::hash::Hasher;
+
+use ahash::AHasher;
+
+use super::HashTableKeyable;
+
+#[derive(Clone, Copy)]
+pub struct KeysRef {
+    pub length: usize,
+    pub address: usize,
+}
```

Comment:
Can src/query/functions-v2/src/aggregates/aggregate_distinct_state.rs also use this struct? It seems there is a duplication.

Reply:
Yes, they are moved there

Reply:
I didn't note this, sorry.